### PR TITLE
Adding NUMBA_ENABLE_PROFILING envvar, enabling jit event

### PR DESCRIPTION
**Pull request by anton-malakhov**
Default value depends on whether profiling was detected
It also serves as a default value for NUMBA_DEBUGINFO

@sklam @seibert

### CODE REVIEW

**Comment by anton-malakhov** [comment]
On Azure, the build error is not related to this PR: `2018-09-13T19:36:47.6095703Z ./miniconda.sh: 373: ./miniconda.sh: /home/vsts/miniconda3/pkgs/python-3.7.0-hc3d631a_0/bin/python: not found`

**Comment by seibert** [comment]
Yeah, this is a random failure that just started happening due to something strange (and apparently random) in the environment setup.  It's almost like some of the build machines on Azure are behaving differently than others.

**Comment by anton-malakhov** [comment]
Ok, I'll try to restart testing by force-pushing this commit

**Comment by seibert** [comment]
Don't worry too much about it for now.  We're treating the Azure results as "informative" while we evaluate how stable the service is.

**Comment by anton-malakhov** [comment]
@seibert could you please review the envvar name and changes to the defaults in this PR? I could sketch a doc section as well, if it works for you this way

**Comment by seibert** [comment]
This looks fine to me, but what code is calling `lib_from_sym` in helperlib?

**Comment by anton-malakhov** [comment]
Oops.. looks like it is a branch collision - I forgot I have this PR open and created this function for solving certain issues with multiple openmp copies. I'll roll it back

File: docs/source/reference/envvars.rst [hunk 1]
```diff
@@ -97,6 +97,14 @@ These variables influence what is printed out during compilation of
    of the compiler frontend, up to and including generation of the Numba
    Intermediate Representation.
 
+.. envvar:: NUMBA_DEBUGINFO
+
+   if sets to non-zero, enable debug for the full application by setting
+   the default value of the ``debug`` option in ``jit``. Beware that
+   enabling debug info significantly increases the memory consumption
+   for each compiled function.
+   Default value euqals to the value of `NUMBA_ENABLE_PROFILING`.
```
Comment (stuartarchibald) [inline docs/source/reference/envvars.rst]: s/euqals/equals/

**Comment by sklam** [comment]
adding reference to https://github.com/numba/llvmlite/pull/394

**Comment by stuartarchibald** [comment]
Testing this PR needs merge of https://github.com/numba/numba/pull/3424, then to do a merge build with master so as to pick up llvmlite 0.26 (where the linked changes are proposed).

**Comment by stuartarchibald** [comment]
Smoke test build passed, only failures were unrelated. Merging!

**Comment by anton-malakhov** [comment]
Thanks!

**Comment by stuartarchibald** [comment]
No problem. Thanks for the contribution!

**Pull request merged**